eZ/Bundle/EzPublishCoreBundle/Command/CopySubtreeCommand.php

```diff
@@ -0,0 +1,162 @@
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+
+class CopySubtreeCommand extends Command
+{
+    /**
+     * @var \Symfony\Component\Console\Input\InputInterface
+     */
+    private $input;
+
+    /**
+     * @var \Symfony\Component\Console\Output\OutputInterface
+     */
+    private $output;
+
+    /**
+     * @var \eZ\Publish\API\Repository\LocationService
+     */
+    private $locationService;
+
+    /**
+     * @var \eZ\Publish\API\Repository\PermissionResolver
+     */
+    private $permissionResolver;
+
+    /**
+     * @var \eZ\Publish\API\Repository\UserService
+     */
+    private $userService;
+
+    /**
+     * @var \eZ\Publish\API\Repository\ContentTypeService
+     */
+    private $contentTypeService;
+
+    /**
+     * @param \eZ\Publish\API\Repository\LocationService $locationService
+     * @param \eZ\Publish\API\Repository\PermissionResolver $permissionResolver
+     * @param \eZ\Publish\API\Repository\UserService $userService
+     * @param \eZ\Publish\API\Repository\ContentTypeService $contentTypeService
+     */
+    public function __construct(
+        LocationService $locationService,
+        PermissionResolver $permissionResolver,
+        UserService $userService,
+        ContentTypeService $contentTypeService
+    ) {
+        parent::__construct();
+        $this->locationService = $locationService;
+        $this->permissionResolver = $permissionResolver;
+        $this->userService = $userService;
+        $this->contentTypeService = $contentTypeService;
+    }
+
+    protected function configure()
+    {
+        $this
+            ->setName('ezplatform:copy-subtree')
+            ->addArgument(
+                'source-location-id',
+                InputArgument::REQUIRED,
+                'Id of subtree root location'
+            )
+            ->addArgument(
+                'target-location-id',
+                InputArgument::REQUIRED,
+                'Id of target location'
+            )
+            ->addOption(
+                'user',
+                'u',
+                InputOption::VALUE_OPTIONAL,
+                'eZ Platform username (with Role containing at least Content policies: read, versionread, edit, remove, versionremove)',
+                'admin'
+            )
+            ->setDescription('Copy subtree from one location to another');
+    }
+
+    /**
+     * @param \Symfony\Component\Console\Input\InputInterface $input
+     * @param \Symfony\Component\Console\Output\OutputInterface $output
+     *
+     * @throws \eZ\Publish\API\Repository\Exceptions\NotFoundException
+     */
+    protected function initialize(InputInterface $input, OutputInterface $output)
+    {
+        parent::initialize($input, $output);
+        $this->input = $input;
+        $this->output = $output;
```

Comment:
These two fields are not needed, since your entire code is in `execute` method. Just use `$output` directly there (right now it's a little bit inconsistent anyway).